inc/spbc-settings.php

```diff
@@ -5144,6 +5144,86 @@ function spbc_show_more_security_logs_callback()
     );
 }
 
+function spbc_show_hostname_security_logs_callback()
+{
+    global $wpdb;
+
+    spbc_check_ajax_referer('spbc_secret_nonce', 'security');
+
+    // validate ips
+    $ips = Post::getArray('ips');
+    if (empty($ips)) {
+        wp_send_json_error('IPs are required.');
+    }
+    $ips = array_map(function ($ip) {
+        return trim($ip);
+    }, $ips);
+    $ips = array_unique($ips);
+    $ips = array_filter($ips, function ($ip) {
+        return IP::validate($ip);
+    });
+    $ips = array_map(function ($ip) {
+        return ip2long($ip);
+    }, $ips);
+
+    // check in db
+    $hostnames = $wpdb->get_results(
+        $wpdb->prepare(
+            "SELECT network, hostname FROM " . SPBC_TBL_SECURITY_LOG_HOSTNAMES . " WHERE network IN (" . implode(',', array_fill(0, count($ips), '%d')) . ")",
+            $ips
+        ),
+        ARRAY_A
+    );
+
+    // separate new ips
+    $new_ips = array_diff($ips, array_column($hostnames, 'network'));
+
+    // Get hostname by IPs using gethostbyaddr
+    if ( ! empty($new_ips) ) {
+        $new_hostnames = array();
+        foreach ($new_ips as $ip) {
+            $new_ip = long2ip($ip);
+            $new_hostname = gethostbyaddr($new_ip);
```

Comment:
Check `gethostbyaddr` existence before

Reply:
There are checking only $new_ips, which not exists in db.

Reply:
Thanks, fixed https://github.com/CleanTalk/security-malware-firewall/pull/507/commits/7568bb047e4b86e30da97c62b4ae02794506976e